### PR TITLE
Refactor attribute selector

### PR DESCRIPTION
**Pull request by alexander-akait**
Fixed:
- Avoid `NamespacedRule`/`TypeSelector` for `name`, because it is invalid
- Rename properties according spec
- Fix logic for invalid attribute matcher
- Fix logic for invalid attribute value
- Simplify
- Handle spaces between `modifier`
- matcher value can be `Ident` (`Text` AST Node) or `Str` (`Str` AST Node)

### CODE REVIEW

**Comment by alexander-akait** [comment]
@kdy1 Ready for review

File: css/ast/src/selector.rs [hunk 1]
```diff
@@ -108,6 +91,34 @@ pub enum AttrSelectorOp {
...
+    #[tag("Text")]
+    Text(Text),
+}
+
+#[ast_node("AttributeSelector")]
+pub struct AttrSelector {
+    pub span: Span,
+    pub prefix: Option<Text>,
+    pub name: Text,
+    pub matcher: Option<AttrSelectorMatcher>,
```
Comment (kdy1) [inline css/ast/src/selector.rs:109]: Is `Matcher` a word used in the specification?

File: css/parser/tests/fixture.rs [hunk 2]
```diff
@@ -317,6 +317,7 @@ impl Visit for SpanVisualizer<'_> {
     mtd!(AtRule, visit_at_rule);
     mtd!(AtSelector, visit_at_selector);
     mtd!(AtTextValue, visit_at_text_value);
+
```
Comment (kdy1) [inline css/parser/tests/fixture.rs:320]: This seems like a mistake?

**Review by kdy1 — COMMENTED** [review]
Code changes look fine to me.
I just have some questions.

File: css/ast/src/selector.rs (hunk 1 above)
Comment (alexander-akait) [reply css/ast/src/selector.rs:109]: Yes

> \<attr-matcher\> = [ '~' | '|' | '^' | '$' | '*' ]? '='

https://www.w3.org/TR/selectors-4/#grammar

File: css/parser/tests/fixture.rs (hunk 2 above)
Comment (alexander-akait) [reply css/parser/tests/fixture.rs]: Yep

File: css/ast/src/selector.rs (hunk 1 above)
Comment (kdy1) [reply css/ast/src/selector.rs:109]: Thanks for the clarification!

**Comment by alexander-akait** [comment]
@kdy1 fixed and ready for review

**Review by kdy1 — APPROVED** [review]
LGTM.
Thank you!

**Pull request merged**